lib/workflows/email/send_decom_flavor_email.py

```diff
@@ -0,0 +1,141 @@
+from typing import List, Optional
+
+from openstack_query import FlavorQuery, UserQuery
+from enums.query.props.flavor_properties import FlavorProperties
+from enums.query.props.server_properties import ServerProperties
+from enums.query.props.project_properties import ProjectProperties
+from enums.query.props.user_properties import UserProperties
+from enums.query.query_presets import QueryPresetsGeneric
+from structs.email.smtp_account import SMTPAccount
+from structs.email.email_template_details import EmailTemplateDetails
+from structs.email.email_params import EmailParams
+from email_api.emailer import Emailer
+from tabulate import tabulate
+
+
+def send_decom_flavor_email(
+    smtp_account: SMTPAccount,
+    cloud_account: str,
+    decom_flavor_list: List[str],
+    project_list: Optional[List[str]] = None,
+    all_projects: bool = False,
+    send_email: bool = False,
+    as_html: bool = False,
+    **kwargs,
+):
+    """
+    Sends an email to each user who owns one or more VMs that have a flavor that is to be decommissioned.
+    This email will contain a notice to delete or rebuild the VM
+    :param smtp_account: (SMTPAccount): SMTP config
+    :param cloud_account: string represents cloud account to use
+    :param decom_flavor_list: A list of flavor names to be decommissioned
+    :param project_list: A list of projects to limit search in
+    :param all_projects: A boolean which if selected will search in all projects
+    :param send_email: Actually send the email instead of printing what will be sent
+    :param as_html: Send email as html
+    :param kwargs: see EmailParams dataclass class docstring
+    """
+
+    if project_list and all_projects:
+        raise RuntimeError(
+            "given both project list and all_projects flag - please choose only one"
+        )
+
+    # flavor query
+    q1 = FlavorQuery().where(
+        QueryPresetsGeneric.EQUAL_TO, FlavorProperties.FLAVOR_IS_PUBLIC, value=True
+    )
+    q1.where(
+        QueryPresetsGeneric.ANY_IN,
+        FlavorProperties.FLAVOR_NAME,
+        values=decom_flavor_list,
+    )
+    q1.run(cloud_account)
+    q1.sort_by((FlavorProperties.FLAVOR_ID, False))
+
+    # server query
+    q2 = q1.then(query_type="SERVER_QUERY", forward_outputs=True).run(
```

Comment:
Ditto on `q2` -> `server_query`, since I want this to be our example we start people with as it has a bit of everything

Reply:
`forward_outputs` feels a bit clunky here, and I can't put my finger on why....

To bikeshed `then ... keep_previous_results = True` makes more sense from just speaking the API
rather than `then forward_results = True` which requires you to explain what a forward is

Reply:
happy with that :+1: